spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractPollableChannel.java

```diff
@@ -123,7 +123,9 @@ else if (logger.isTraceEnabled()) {
 				logger.trace("postReceive on channel '" + this + "', message is null");
 			}
 			if (!CollectionUtils.isEmpty(interceptorStack)) {
-				message = interceptorList.postReceive(message, this);
+				if (message != null) {
+					message = interceptorList.postReceive(message, this);
```

Comment:
Can't we follow a similar logic in the `PollableAmqpChannel` or `PollableJmsChannel` like this:
```
if (object == null) {
		if (logger.isTraceEnabled()) {
				logger.trace("postReceive on channel '" + this + "', message is null");
		}
		return null;
}
```

I mean instead of check here just return `null` after trace log message.

Reply:
No because we need to call `afterReceiveCompleted()` - its API allows `null`.

Reply:
So, `PollableAmqpChannel` and `PollableJmsChannel` have to be fixed respectively?
They just `return null;` and definitely don't call that `afterReceiveCompleted()`...

Reply:
OK. So, I'm leaving that for you to fix in the subsequent commits.

I'm not sure that we are going to be good to back-port it, since in the end it will be as a behavior change.

Reply:
Right - no backport.